packages/expo-video/android/src/main/java/expo/modules/video/VideoRecords.kt

```diff
@@ -0,0 +1,65 @@
+package expo.modules.video
+
+import androidx.media3.common.C
+import androidx.media3.common.MediaItem
+import expo.modules.kotlin.records.Field
+import expo.modules.kotlin.records.Record
+import expo.modules.kotlin.types.Enumerable
+import java.io.Serializable
+import java.util.UUID
+
+internal class VideoSource(
+  @Field var uri: String? = null,
+  @Field var drm: DRMArguments? = null
+) : Record, Serializable {
+  fun toMediaItem(): MediaItem {
+    val mediaItem = MediaItem
+      .Builder()
+      .setUri(this.uri ?: "")
+
+    this.drm?.let {
+      if (it.type.isSupported()) {
+        mediaItem.setDrmConfiguration(it.toDRMConfiguration())
+      } else {
+        throw UnsupportedDRMTypeException(it.type)
+      }
+    }
+    return mediaItem.build()
+  }
+}
+
+internal enum class DRMType(val value: String) : Enumerable {
+  CLEARKEY("clearkey"),
+  FAIRPLAY("fairplay"),
+  PLAYREADY("playready"),
+  WIDEVINE("widevine");
+
+  fun isSupported(): Boolean {
+    return this != FAIRPLAY
+  }
+
+  fun toUUID(): UUID {
+    return when (this) {
+      CLEARKEY -> C.CLEARKEY_UUID
+      FAIRPLAY -> throw UnsupportedDRMTypeException(this)
+      PLAYREADY -> C.PLAYREADY_UUID
+      WIDEVINE -> C.WIDEVINE_UUID
+    }
+  }
+}
+
+internal class DRMArguments(
```

Comment:
```suggestion
internal class DRMOptions(
```

Also, it might be a good idea to split these three classes to separate files.